src/processor/operator/persistent/reader/csv/base_csv_reader.cpp

```diff
@@ -78,10 +80,14 @@ bool BaseCSVReader::addValue(Driver& driver, uint64_t rowNum, column_id_t column
         }
         newVal += strVal.substr(prevPos, strVal.size() - prevPos);
         escapePositions.clear();
-        return driver.addValue(rowNum, columnIdx, newVal);
+        valueToAdd = newVal;
     } else {
-        return driver.addValue(rowNum, columnIdx, strVal);
+        valueToAdd = strVal;
     }
+    if (!utf8proc::Utf8Proc::isValid(valueToAdd.data(), valueToAdd.length())) {
```

Comment:
Does this imply we will be noticeable slower when reading csv after this PR?

Reply:
I think we won't be noticeable slower. This is necessary step, we should validate every value in csv before processing it.